src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -84,11 +84,13 @@ trait TreeAndTypeAnalysis extends Debugging {
     tp <:< tpImpliedNormalizedToAny
   }
 
-  // TODO: improve, e.g., for constants
-  def sameValue(a: Tree, b: Tree): Boolean = (a eq b) || ((a, b) match {
-    case (_ : Ident, _ : Ident) => a.symbol eq b.symbol
-    case _                      => false
-  })
+  def equivalentPath(a: Tree, b: Tree): Boolean = (a, b) match {
+    case (Select(qual1, _), Select(qual2, _)) => equivalentPath(qual1, qual2) && a.symbol == b.symbol
```

Comment:
I checked that we covered all cases covered by compiling the standard library. But I'll try some variations manually and see if we can get here with an `Apply`.